learn/getting_started/algolia_migration.md

```diff
@@ -0,0 +1,292 @@
+
+```bash
+mkdir algolia-meilisearch-migration
+cd algolia-meilisearch-migration
+npm init
+```
+
+The `npm init` command should walk you through creating a `package.json` file within your project. This file contains details about the project's packages and dependencies, plus some basic metadata. You can press `^C` at any time to exit.
+
+Next, create a `script.js` file:
+
+```bash
+touch script.js
+```
+
+This is where we will add the migration code.
+
+### Install required packages
+
+To get started, we'll need two different packages. The first is `algoliasearch`, the JavaScript client for the Algolia API, and the second is `meilisearch`, the JavaScript client for the Meilisearch API.
+
+```bash
+npm install algoliasearch meilisearch
+```
+
+### Create Algolia client
+
+You'll need your **Application ID** and **API Key** to start the Algolia client. Both can be found in your [Algolia account](https://www.algolia.com/api-keys).
+
+Paste the below code in `script.js`:
+
+```jsx
+const AlgoliaSearch = require("algoliasearch");
+const algoliaClient = AlgoliaSearch(
+  "APPLICATION_ID",
+  "ADMIN_API_KEY"
+);
+const algoliaIndex = algoliaClient.initIndex("INDEX_NAME");
+```
+
+Replace `APPLICATION_ID` and `ADMIN_API_KEY`with your Algolia application ID and admin API key respectively.
+
+Replace `INDEX_NAME` with the name of the Algolia index you would like to migrate to Meilisearch.
+
+### Fetch data from Algolia
+
+To fetch all index data at once we can use Algolia's [`browseObjects`](https://www.algolia.com/doc/api-reference/api-methods/browse/) method.
+
+```jsx
+let records = [];
+await algoliaIndex.browseObjects({
+    batch: (hits) => {
+      records=records.concat(hits)
+    }
+  });
+```
+
+On each batch of hits, the `batch` callback method is invoked, and the content is concatenated in the `records` array. This variable will be used later on in the upload process.
+
+#### Nested objects
+
+Meilisearch versions prior to v0.27.0 have limited support for nested fields, meaning that nested objects must be flattened manually. For this reason we recommend using [the latest version of Meilisearch](https://github.com/meilisearch/MeiliSearch/releases). If this is impossible, try the modified data retrieval process below.
+
+Let's say your document structure looks like this:
+
+```json
+{
+    "objectID": 1,
+    "name": "John",
+    "age": 20,
+    "address": {
+        "country": "Spain"
+    }
+}
+```
+
+To export this data from Algolia and simultaneously flatten it, we will use the same `browseObjects` method as described above. However, before concatenation, we will modify the object to flatten `address` into `address.country`:
+
+```jsx
+let records = [];
+await algoliaIndex.browseObjects({
+ batch: (recordsArr) => {
+        records = recordsArr.reduce((acc, record) => {
+          const { address, ...rest } = record;
+          acc.push({ "address.country": address.country, ...rest });
```

Comment:
I think @bidoubiwa's solution is technically more complete and closer to real-life scenarios where data won't always conveniently formatted. But it is also more complex and risks distracting users who are more concerned about understanding the workflow than copypasting code.

Reply:
We discussed with @shivaylamba and decided to remove this part altogether. We discussed this subject a lot in the past and determined back then that we should probably not keep support for previous versions of Meilisearch (before the v1). We don't want users to keep older versions for their production because for example, older versions might have bugs that Meilisearch (the engine) will not fix.

It also adds a lot of complexity to this article for a very specific case. Nested is supported in v0.27.1, if the user wants I think that we should invite him to upgrade to this version.

Reply:
I agree with this decision! Thanks @bidoubiwa and @shivaylamba 😊